lib/hooks/slides.tsx

```diff
@@ -0,0 +1,79 @@
+import { isNumber } from "lodash-es";
+import { useEffect, useRef, useState } from "react";
+
+export type UseSlidesProps = {
+  count: number;
+  autoplay?: number;
+  currentSlide?: number;
+  pauseOnUserInteraction?: number;
+};
+
+export type UseSliderControls = {
+  currentSlide: number;
+  next: (userOrigin?: boolean) => void;
+  prev: (userOrigin?: boolean) => void;
+  goto: (slide: number, userOrigin?: boolean) => void;
+  pause: (time: number) => void;
+};
+
+export const useSliderControls = (props: UseSlidesProps): UseSliderControls => {
+  const [currentSlide, setCurrentSlide] = useState<number>(
+    props.currentSlide ?? 0,
+  );
+
+  const pauseTimerRef = useRef<NodeJS.Timeout | null>(null);
+  const [paused, setPaused] = useState(false);
+
+  const prev = (userOrigin?: boolean) => {
```

Comment:
I think the names of the function are reversed: `prev` seems to move to slider forward and vice-versa with `next`

The autoplay function didn't seem to work correctly but when I switched the names it seemed to work just fine.

Reply:
Ah yeah! Good catch haha.